### ISSUE TITLE
Don't nuke directory specified in out.dir

### ISSUE THREAD

**Issue by man-shar** [tdhock/animint2#86]
Just for a quickly trying out `out.dir`, I put in a directory I was using and thought it would just create a few files I can later delete.

Unfortunately ended up losing a few files :P

Might be better just adding to the folder rather than nuking it?

**Comment by tdhock** [comment]
Sorry to hear that you lost some files, but this behavior is documented in ?animint2dir, so I don't think that any change is warranted at this time,
```
\item{out.dir}{directory to store html/js/csv files. If it exists
already, it will be removed before writing the new
directory/files.}
```

**Comment by tdhock** [comment]
At some point in the past we did do that (adding to the folder rather than nuking it) but it resulted in some difficult issues when debugging, so that is how we landed on the current approach. Is there some other place where you think we could mention this behavior?

On https://rcdata.nau.edu/genomic-ml/animint2-manual/Ch05-sharing.html is says:
If the parent of out.dir does not exist, that is an error (you can use dir.create to create the parent if necessary). If out.dir does not exist, then it will be created. If out.dir does exist, then any files in that directory will be overwritten.

**Comment by man-shar** [comment]
Hmmm, I understand. That documentation seems appropriate to me.

In my own experience with dev servers, there seems to be a variation in how they handle it. [parcel](https://github.com/parcel-bundler/parcel) doesn't empty the folder, while [vite](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/build.ts#L95-L100) chooses to empty it by default, unless `emptyOutDir` config option is set to `false`.

Parcel's creator mentions my particular use case where the build and work directory could be the same in this [reply](https://github.com/parcel-bundler/parcel/issues/3591#issuecomment-539567824).

This might just boil down to personal taste then I suppose. What I feel would be safer is:
1. If `out.dir` isn't specified, do the usual temp directory creation.
2. If `out.dir` _is_ specified, create an `animint-dist/` or `animint-build` folder inside `out.dir`, and use that to run the server.

Either decision is fine with me tbh – I don't have much experience with R libraries.

**Comment by tdhock** [comment]
fixed via an error message #88
